lib/sycamore/sycamore/document.py

```diff
@@ -0,0 +1,390 @@
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_section_header(self)
+
+
+class Table(Leaf):
+    def __init__(self, node_id, box, content, properties, continued=None):
+        super().__init__(node_id, box, content, properties)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_table(self)
+
+
+class Text(Leaf):
+    def __init__(self, node_id, box, content, properties, continued=None):
+        super().__init__(node_id, box, content, properties)
+        self._continued = continued
+
+    def category(self):
+        return Category.Caption
+
+    def continued(self):
+        return self._continued
+
+    def accept(self, visitor):
+        return visitor.visit_text(self)
+
+
+class Title(Leaf):
+    def __init__(self, node_id, box, content, properties):
+        super().__init__(node_id, box, content, properties)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_title(self)
+
+
+class Group(Internal):
+    """
+    Group semantic related objects together, e.g. list items, figure and caption
+    """
+
+    def __init__(self, node_id, children, properties):
+        super().__init__(node_id, children, properties)
+
+    def category(self):
+        return Category.Group
+
+    def accept(self, visitor):
+        return visitor.visit_group(self)
+
+
+class Section(Internal):
+    def __init__(self, node_id, children, header, properties):
+        super().__init__(node_id, children, properties)
+        self._header = header
+
+    def category(self):
+        return Category.Section
+
+    def accept(self, visitor):
+        return visitor.visit_section(self)
+
+
+class Visitor(ABC):
+    @abstractmethod
+    def visit_caption(self, caption: Caption):
+        pass
+
+    @abstractmethod
+    def visit_footnote(self, footnote: Footnote):
+        pass
+
+    @abstractmethod
+    def visit_formula(self, formula: Formula):
+        pass
+
+    @abstractmethod
+    def visit_list_item(self, list_item: ListItem):
+        pass
+
+    @abstractmethod
+    def visit_page_footer(self, page_footer: PageFooter):
+        pass
+
+    @abstractmethod
+    def visit_page_header(self, page_header: PageHeader):
+        pass
+
+    @abstractmethod
+    def visit_picture(self, picture: Picture):
+        pass
+
+    @abstractmethod
+    def visit_section_header(self, section_header: SectionHeader):
+        pass
+
+    @abstractmethod
+    def visit_table(self, table: Table):
+        pass
+
+    @abstractmethod
+    def visit_text(self, text: Text):
+        pass
+
+    @abstractmethod
+    def visit_title(self, title: Title):
+        pass
+
+    @abstractmethod
+    def visit_group(self, group: Group):
+        pass
+
+    @abstractmethod
+    def visit_section(self, section: Section):
+        pass
+
```

Comment:
I feel like a visit_generic or something would be useful. Something like
```
@abstractmethod
def visit_generic(self, node: Node):
    pass
```
and then the `visit_some_type` methods can default to calling visit_generic. I'm just looking at these and thinking it will be an absolute pain to have to explicitly handle each type when I wanna implement this interface ya know? Could save a bunch of code duplication in the NaiveSemanticVisitor eg